packages/hop-node/src/db/TransferRootsDb.ts

```diff
@@ -195,212 +202,106 @@ class TransferRootsDb extends BaseDb {
     }
+  }
 
-    await Promise.all(promises)
+  private async upsertTimestampedKeyItem (transferRoot: TransferRoot) {
+    const { transferRootId } = transferRoot
+    const logger = this.logger.create({ id: transferRootId })
+    const key = this.getTimestampedKey(transferRoot)
+    if (key) {
+      const exists = await this.subDbTimestamps.getById(key)
+      if (!exists) {
+        logger.debug(`storing db transferRoot timestamped key item. key: ${key}`)
+        await this.subDbTimestamps._update(key, { transferRootId })
+        logger.debug(`updated db transferRoot timestamped key item. key: ${key}`)
+      }
+    }
+  }
 
-    // re-index db keys by transfer root id
-    items = await this.getKeyValues()
-    promises = []
-    for (const { key, value } of items) {
-      promises.push(new Promise(async (resolve) => {
-        if (!value?.transferRootId || key === value?.transferRootId) {
-          resolve(null)
-          return
-        }
-        const { transferRootId } = value
-        await this._update(transferRootId, value)
-        await this.deleteById(key)
-        resolve(null)
-      }))
+  private async upsertRootHashKeyItem (transferRoot: TransferRoot) {
+    const { transferRootId, transferRootHash } = transferRoot
+    const logger = this.logger.create({ id: transferRootId })
+    const key = transferRoot.transferRootHash
+    if (key) {
+      const exists = await this.subDbRootHashes.getById(key)
+      if (!exists) {
+        logger.debug(`storing db transferRoot rootHash key item. key: ${key}`)
+        await this.subDbRootHashes._update(key, { transferRootId })
+        logger.debug(`updated db transferRoot rootHash key item. key: ${key}`)
+      }
     }
+  }
 
-    await Promise.all(promises)
+  private async upsertTransferRootItem (transferRoot: TransferRoot) {
+    const { transferRootId } = transferRoot
+    const logger = this.logger.create({ id: transferRootId })
+    await this._update(transferRootId, transferRoot)
+    const entry = await this.getById(transferRootId)
+    logger.debug(`updated db transferRoot item. ${JSON.stringify(entry)}`)
+    await this.updateIncompleteItem(entry)
   }
 
-  async updateIncompleteItem (item: Partial<TransferRoot>) {
-    if (!item) {
-      this.logger.error('expected item', item)
-      return
-    }
-    const { transferRootId } = item
-    if (!transferRootId) {
-      this.logger.error('expected transferRootId', item)
-      return
-    }
-    const isIncomplete = this.isItemIncomplete(item)
+  private async updateIncompleteItem (transferRoot: TransferRoot) {
+    const { transferRootId } = transferRoot
+    const logger = this.logger.create({ id: transferRootId })
+    const isIncomplete = this.isItemIncomplete(transferRoot)
     const exists = await this.subDbIncompletes.getById(transferRootId)
     const shouldUpsert = isIncomplete && !exists
     const shouldDelete = !isIncomplete && exists
+    logger.debug('storing db transferRoot incomplete key item')
```

Comment:
nitpick: it is possible that we do not store anything here. this log should be moved within the if statement or reworded.

Reply:
Good idea. The debug logs have moved and been updated 

Resolved with https://github.com/hop-protocol/hop/pull/351/commits/6fd88c22d80439b9ab8349efbb41c7b7f8b5b1ec